upstream-community-operators/postgres-operator/postgres-operator.v1.2.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,340 @@
+      mediatype: image/png
+  links:
+  - name: Documentation
+    url: https://postgres-operator.readthedocs.io/en/latest/
+  - name: GitHub
+    url: https://github.com/zalando/postgres-operator
+  customresourcedefinitions:
+    owned:
+    - name: postgresqls.acid.zalan.do
+      version: v1
+      kind: postgresql
+      displayName: PostgreSQL
+      description: Creates a PostgreSQL cluster managed by Postgres-Operator.
+  installModes:
+  - type: OwnNamespace
+    supported: true
+  - type: SingleNamespace
+    supported: false
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+      - serviceAccountName: zalando-postgres-operator
+        rules:
+        - apiGroups:
+          - "rbac.authorization.k8s.io"
+          resources:
+          - clusterrolebindings
+          - clusterroles
+          verbs:
+          - get
+          - create
+          - delete
+          - patch
+          - list
+      permissions:
+      - serviceAccountName: zalando-postgres-operator
+        rules:
+        - apiGroups:
+          - acid.zalan.do
+          resources:
+          - postgresqls
+          - postgresqls/status
+          - operatorconfigurations
+          verbs:
+          - "*"
+        - apiGroups:
```

Comment:
Your Operator tries to create CRDs upon start-up which fails because CRDs are cluster-scoped resources. Move this section up in the `clusterPermissions` section to always have this RBAC be at cluster-scope.
The test deploys your Operator to watch all namespaces, causing all your RBAC to be elevated to cluster permissions - that's why the test does not fail to deploy your Operator.

Reply:
Alternatively remove the `ownNamespace` InstallMode.

Reply:
I moved the CRD RBAC part to `clusterPermissions`